src/app/views/uploads/dataset/upload-details/DatasetUploadDetails.jsx

```diff
@@ -12,141 +12,98 @@ import http from "../../../../utilities/http";
+            const activeDatasetAPIResponse = mapAPIResponsesToState({ recipe, job });
 
-            if (!activeDataset) {
+            if (!activeDatasetAPIResponse) {
                 const notification = {
                     type: "neutral",
                     message: "This dataset was not recognised, so you've been redirected to the main screen",
                     isDismissable: true,
                 };
                 notifications.add(notification);
-                this.props.dispatch(push(url.resolve("../")));
+                props.dispatch(push(url.resolve("../")));
                 return;
             }
-
-            this.setState({ activeDataset });
-        }
-    }
-
-    componentDidMount() {
-        if (!this.state.activeDataset) {
-            return;
+            setActiveDataset({ activeDatasetAPIResponse });
+
+            if (activeDataset.status === "completed" && !activeDataset.dimensions) {
+                datasetImport.getDimensions(activeDataset.instanceID).then(response => {
+                    const activeDatasetWithDimensions = {
+                        ...activeDataset,
+                        dimensions: [],
+                    };
+                    response.map(dimension => {
+                        activeDatasetWithDimensions.dimensions.push(dimension.name);
+                    });
+                    setActiveDataset({ activeDatasetWithDimensions });
+                });
+            }
+            bindInputs();
         }
+    }, []);
 
-        this.bindInputs();
-    }
+    const repeatUploadStatusCheck = () => {
+        setIsFetchingDataset(true);
+        setLoadingPageForFirstTime(true);
+        getUploadStatus();
 
-    componentDidUpdate() {
-        if (!this.state.activeDataset) {
-            return;
-        }
-
-        if (this.state.activeDataset.status === "completed" && !this.state.activeDataset.dimensions) {
-            datasetImport.getDimensions(this.state.activeDataset.instanceID).then(response => {
-                const activeDataset = {
-                    ...this.state.activeDataset,
-                    dimensions: [],
-                };
-                response.map(dimension => {
-                    activeDataset.dimensions.push(dimension.name);
-                });
-                this.setState({ activeDataset });
-            });
-        }
-        this.bindInputs();
-    }
-
-    repeatUploadStatusCheck = () => {
-        this.setState(
-            {
-                isFetchingDataset: true,
-                loadingPageForFirstTime: true,
-            },
-            () => {
-                this.getUploadStatus();
-            }
-        );
-        this.intervalID = setInterval(async () => {
-            if (!this.state.isFetchingDataset) {
-                this.setState({ isFetchingDataset: true }, () => {
-                    this.getUploadStatus();
-                });
+        intervalID = setInterval(async () => {
```

Comment:
IntervalID is declared above as a const either this is a new variable in which case we should explicitly state it or the const should be a `let` on line 21